spikeextractors/baseextractor.py

```diff
@@ -32,47 +39,47 @@ def make_serialized_dict(self):
         if self.is_dumpable:
             dump_dict = {'class': class_name, 'module': module, 'kwargs': self._kwargs,
                          'version': imported_module.__version__, 'dumpable': True}
-            if 'Recording' in class_name:
-                if 'group' in self.get_shared_channel_property_names():
-                    groups = self.get_channel_groups()
-                    dump_dict['groups'] = groups
-                if 'location' in self.get_shared_channel_property_names():
-                    locations = self.get_channel_locations()
-                    dump_dict['locations'] = locations
-            elif 'Sorting' in class_name:
-                if 'group' in self.get_shared_unit_property_names():
-                    groups = [self.get_unit_property(u, 'group') for u in self.get_unit_ids()]
-                    dump_dict['groups'] = groups
         else:
             dump_dict = {'class': class_name, 'module': module, 'kwargs': {},
                          'version': imported_module.__version__, 'dumpable': False}
-
+        dump_dict = self._include_properties_and_features_in_dict(dump_dict, include_properties, include_features)
         return dump_dict
 
-    def dump(self, file_name=None, folder_path=None):
+    def dump(self, file_path=None, dump_properties_and_features=True):
```

Comment:
I think I would put the same signautre than ```make_serialized_dict``` with include_properties=None, include_features=None.
But by default I would put include_properties='all' and 'all would be all feature or properties.